src/coreclr/vm/threads.cpp

```diff
@@ -8459,6 +8459,11 @@ Thread::EnumMemoryRegionsWorker(CLRDataEnumMemoryFlags flags)
     }
     else
     {
+        // Skip any thread that doesn't have a OS thread id because DacGetThreadContext is going to throw an exception.
+        if (GetOSThreadId() == 0)
+        {
+            return;
```

Comment:
Now that we have a logging mechanism should we log a message about this?

Reply:
This seems like a kind of normal condition to have no native thread id.  This check was to keep the GetThreadContext from throwing and logging a message. I wanted to keep the "exception thrown" logging only for actually problems.

So I don't think it needs logging.